wasi/wasi.go

```diff
@@ -293,3 +293,25 @@ var errnoToString = [...]string{
 	ErrnoXdev:           "EXDEV",
 	ErrnoNotcapable:     "ENOTCAPABLE",
 }
+
+// ExitCode is an arbitrary uint32 number returned by proc_exit.
+// An exit code of 0 indicates successful termination. The meanings of other values are not defined by WASI.
+//
+// In wazero, if ProcExit is called, the calling function returns immediately, returning the given exit code as the error.
+// You can get the exit code by casting the error as follows.
+//
+//   wasmFunction := m.GetFunctionVoidReturn(/ * omitted */)  // Some function which may call proc_exit
+//   err := wasmFunction()
+//   var exitCode wasi.ExitCode
+//   if errors.As(err, &exitCode) {
+//   	// The function is terminated by proc_exit.
+//   }
+//
+// See https://github.com/WebAssembly/WASI/blob/main/phases/snapshot/docs.md#exitcode
+// See https://github.com/WebAssembly/WASI/blob/main/phases/snapshot/docs.md#proc_exit
+type ExitCode uint32
+
+// Error returns the string representation of ExitCode.
```

Comment:
I think we don't need this as `Error() string` is a widely known general interface  https://github.com/golang/go/blob/master/src/builtin/builtin.go#L268-L272
```suggestion
```